### PR TITLE
socket-proxyd: Introduced dynamic connection limit via an option.

### PR DESCRIPTION
**Pull request by galaxy4public**
This PR allows one to dynamically specify the connection limit, since the default of 256 is too restrictive in some cases.

### CODE REVIEW

File: man/systemd-socket-proxyd.xml [hunk 1]
```diff
@@ -85,6 +85,12 @@
     <variablelist>
       <xi:include href="standard-options.xml" xpointer="help" />
       <xi:include href="standard-options.xml" xpointer="version" />
+      <varlistentry>
+        <term><option>--connections-max</option></term>
```
Comment (poettering) [inline man/systemd-socket-proxyd.xml]: Hmm, so, the respective options in .socket units is called `MaxConnections=`, hence I'd propose we call this option `--max-connections=`, too, just in order to stick to one nomenclature.

Also, we usually put a "=" at the end of command line switches that expect an argument in the docs, to clarify that they expect an argument.

File: man/systemd-socket-proxyd.xml [hunk 2]
```diff
@@ -85,6 +85,12 @@
...
+        <term><option>-c</option></term>
+
+        <listitem><para>Sets the maximum number of simultaneous connections.  The default is 256.</para></listitem>
```
Comment (poettering) [inline man/systemd-socket-proxyd.xml]: I think this should also say: "If the limit of concurrent connections is reached further connections will be refused." or something like that.

File: src/socket-proxy/socket-proxyd.c [hunk 3]
```diff
@@ -563,6 +564,7 @@ static void help(void) {
         printf("%1$s [HOST:PORT]\n"
                "%1$s [SOCKET]\n\n"
                "Bidirectionally proxy local sockets to another (possibly remote) socket.\n\n"
+               "  -c --connections-max   Set the maximum number of connections to be accepted\n"
```
Comment (poettering) [inline src/socket-proxy/socket-proxyd.c]: In the help text please also append "=" to the switch to indicate that this expects an argument.

File: src/socket-proxy/socket-proxyd.c [hunk 4]
```diff
@@ -589,7 +589,7 @@ static int parse_argv(int argc, char *argv[]) {
         assert(argc >= 0);
         assert(argv);
 
-        while ((c = getopt_long(argc, argv, "h", options, NULL)) >= 0)
+        while ((c = getopt_long(argc, argv, "c:h", options, NULL)) >= 0)
```
Comment (poettering) [inline src/socket-proxy/socket-proxyd.c:592]: please merge the two changes into one.

**Comment by galaxy4public** [comment]
Unfortunately, I'm not that good with Git and GitHub, so I failed to cleanup the history of the original PR to comply to your request of merging two commits into one.  Hence I submitted a new PR #4749 which incorporates all the requested changes and is clean.

**Pull request closed without merging**